scopes/compilation/compiler/types.ts

```diff
@@ -50,15 +56,23 @@ export interface Compiler extends CompilerOptions {
    */
   id: string;
 
+  icon?: string;
+
   /**
    * serialized config of the compiler.
    */
   displayConfig?(): string;
 
   /**
-   * transpile a single file. this being used during development and get saved into the workspace
+   * transpile a single file that gets saved into the workspace, used by `bit compile` and during
+   * development
+   */
+  transpileFile?: (fileContent: string, params: TranspileFileParams) => TranspileFileOutput;
+
+  /**
+   * transpile all the files of a component, use this when you can't use `transpileFile`
    */
-  transpileFile: (fileContent: string, options: TranspileOpts) => TranspileOutput;
+  transpileComponent?: (component: ConsumerComponent, params: TranspileComponentParams) => Promise<void>;
```

Comment:
no relevant data we can return back here?

Reply:
we could at least return errors if any

Reply:
Maybe a boolean for success ?